packages/terra-worklist-data-grid/src/ColumnHeaderCell.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { injectIntl } from 'react-intl';
+import ThemeContext from 'terra-theme-context';
+import IconUp from 'terra-icon/lib/icon/IconUp';
+import IconDown from 'terra-icon/lib/icon/IconDown';
+import IconError from 'terra-icon/lib/icon/IconError';
+import WorklistDataGridPropTypes from './proptypes/WorklistDataGridPropTypes';
+import './_elementPolyfill';
+import styles from './ColumnHeaderCell.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Data for columns. By default, columns will be presented in the order given.
+   */
+  column: WorklistDataGridPropTypes.columnShape.isRequired,
+  /**
+   * String that specifies the default width for columns in the grid. Any valid CSS width value is accepted.
+   */
+  width: PropTypes.string.isRequired,
+  /**
+   * String that specifies the column height. Any valid CSS height value accepted.
+  */
+  headerHeight: PropTypes.string.isRequired,
+  /**
+   * Function that is called when a selectable header cell is selected. Parameters: `onColumnSelect(columnId)`
+   */
+  onColumnSelect: PropTypes.func,
+  /**
+   * @private
+   * Object containing intl APIs
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }),
+};
+
+const ColumnHeaderCell = (props) => {
+  const {
+    column,
+    width,
+    headerHeight,
+    onColumnSelect,
+    intl,
+  } = props;
+
+  let sortIndicatorIcon;
+  let errorIcon;
+
+  // Add error icon when column error exists
+  if (column.hasError) {
+    errorIcon = <IconError a11yLabel={intl.formatMessage({ id: 'Terra.worklistDataGrid.columnError' })} className={cx('error-icon')} />;
+  }
+
+  // Add the sort indicator based on the sort direction
+  if (column.sortIndicator === WorklistDataGridPropTypes.SortIndicators.ASCENDING) {
+    sortIndicatorIcon = <IconUp />;
+  } else if (column.sortIndicator === WorklistDataGridPropTypes.SortIndicators.DESCENDING) {
+    sortIndicatorIcon = <IconDown />;
+  }
+
+  // Retrieve current theme from context
+  const theme = useContext(ThemeContext);
+
+  return (
+  /* eslint-disable react/forbid-dom-props */
+    <th
+      key={column.id}
+      className={cx('column-header', theme.className, { selectable: !(column.isSelectable === false) })}
+      tabIndex="-1"
+      role="columnheader"
+      scope="col"
+      aria-sort={column.sortIndicator}
+      {...(!(column.isSelectable === false) && onColumnSelect && { onClick: () => { onColumnSelect(column.id); } })}
```

Comment:
Is there a difference between doing this and doing something like:
```suggestion
      onClick={(column.isSelectable && onColumnSelect) ? handleColumnSelect : undefined }
```

And then define `handleColumnSelect` to be something like this in terra-clinical-data-grid: https://github.com/cerner/terra-clinical/blob/main/packages/terra-clinical-data-grid/src/subcomponents/Cell.jsx#L100-L106

Where onClick is defined as: https://github.com/cerner/terra-clinical/blob/main/packages/terra-clinical-data-grid/src/subcomponents/Cell.jsx#L167